config/state.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"os"
+	"regexp"
+
+	"github.com/BurntSushi/toml"
+)
+
+type mixSection struct {
+	Format string `toml:"FORMAT"`
+}
+
+// MixState holds the current state of the mix
+type MixState struct {
+	Mix mixSection
+
+	/* hidden properties */
+	filename string
+	version  string
+}
+
+// CurrentStateVersion is the current revision for the state file structure
+var CurrentStateVersion = "1.0"
```

Comment:
Does this override or get overwritten by the "1.0" in config.go ? `CurrentConfigVersion` specifically*

Reply:
They serve the same purpose but for different files. Config has a version and State has a version. They are not tied, they just happen to be both at version 1.0 right now. I'm not sure how useful it will be for the state file, but since I now see that It would be much better if builder.conf had it from the start, I'm also adding it for the state file.